nautobot/cloud/tests/test_filters.py

```diff
@@ -28,3 +33,28 @@ def setUpTestData(cls):
         for i, _ in enumerate(cls.cloud_accounts):
             cls.cloud_accounts[i].secrets_group = secrets_groups[i]
             cls.cloud_accounts[i].validated_save()
+
+
+class CloudTypeTestCase(FilterTestCases.NameOnlyFilterTestCase):
+    queryset = CloudType.objects.all()
+    filterset = CloudTypeFilterSet
+    generic_filter_tests = [
+        ("description",),
+        ("name",),
+        ("provider", "provider__id"),
+        ("provider", "provider__name"),
+    ]
+
+    def test_content_types(self):
+        cts = [
+            ContentType.objects.get_for_model(Device),
+            ContentType.objects.get_for_model(RackGroup),
+        ]
+        for idx, cloud_type in enumerate(CloudType.objects.all()[:5]):
+            cloud_type.content_types.add(cts[idx % 2])
+
+        params = {"content_types": ["dcim.device", "dcim.rackgroup"]}
+        self.assertEqual(
+            self.filterset(params, self.queryset).qs.count(),
+            CloudType.objects.filter(Q(content_types__in=cts[0])).filter(Q(content_types__in=cts[1])).count(),
```

Comment:
This *feels* over complex to me - would this work instead?

```suggestion
            CloudType.objects.filter(content_types=cts[0]).filter(content_types=cts[1]),
```